easybuild/toolchains/linalg/intelmkl.py

```diff
@@ -84,6 +85,7 @@ def _set_blas_variables(self):
         interfacemap = {
             TC_CONSTANT_INTELCOMP: 'intel',
             TC_CONSTANT_GCC: 'gf',
+            TC_CONSTANT_PGI: 'intel',
```

Comment:
based on https://software.intel.com/en-us/articles/intel-mkl-link-line-advisor, we should always link with `libmkl_intel_lp64.a` when using Intel or PGI compilers, and also for GCC C/C++; only with GCC Fortran `libmkl_gf_lp64.a` should be used instead

Also, `libmkl_intel_thread.a` should always be used, regardless of the compiler, even for GCC or PGI (which is not what we're doing now). I wonder why `libmkl_pgi_thread.a` and `libmkl_gnu_thread.a` are there at all then.

This is based on the advice provided for imkl 11.3.2 specifically...